llmc/cuda_utils.cuh

```diff
@@ -106,7 +106,15 @@ __global__ void copy_and_cast_kernel(Td* dst, const Ts* src, size_t n, ptrdiff_t
     int idx = blockIdx.x * blockDim.x + threadIdx.x;
     // need to try grid stride looping for more perf later
     if (idx < n) {
-        dst[idx + stride_dst * blockIdx.y] = cast_value<Td, Ts>(src[idx + stride_src * blockIdx.y]);
+        dst[idx + stride_dst * blockIdx.y] = cast_value<Td>(src[idx + stride_src * blockIdx.y]);
```

Comment:
we don't need to be explicit with Ts here or why remove this, since I see that the cast_value is "templatized" with both Td and Ts

Reply:
yes, Ts is uniquely determined by the function arguments; whereas Td cannot be inferred, since it only appears in the return type